EIPS/eip-2029.md

```diff
@@ -0,0 +1,61 @@
+---
+eip: 2029
+title: State Rent A - State counters contract
+author: Alexey Akhunov (@AlexeyAkhunov)
+discussions-to: https://ethereum-magicians.org/t/eip-2029-state-counters-contract-change-a-from-state-rent-v3-proposal/3279
+status: Draft
+type: Standards Track
+category: Core
+created: 2019-05-15
+---
+
+<!--You can leave these HTML comments in your merged EIP and delete the visible duplicate text guides, they will not appear and may be helpful to refer to if you edit it again. This is the suggested template for new EIPs. Note that an EIP number will be assigned by an editor. When opening a pull request to submit your EIP, please use an abbreviated title in the filename, `eip-draft_title_abbrev.md`. The title should be 44 characters or less.-->
+
+## Simple Summary
+<!--"If you can't explain it simply, you don't understand it well enough." Provide a simplified and layman-accessible explanation of the EIP.-->
+A smart contract is deployed on all Ethereum networks (including testnets), with the code that simply reads the slot in its storage specified by the
+only parameter. Later, this contract becomes "special" in that Ethereum start writing state counters (number of total transactions, number of accounts,
+etc.) into that contract.
+
+## Abstract
+<!--A short (~200 word) description of the technical issue being addressed.-->
+It is part of the State Rent roadmap. This particular change introduces a place in the Ethereum state where various state counters can be stored. At this
+point, the most important counter is the total number of transactions happened, and this counter will be used to populate the nonces of newly created
+non-contract accounts. This way of populating nonce ensures replay protection for accounts that were evicted and then brought back by sending ether to them.
+
+## Motivation
+<!--The motivation is critical for EIPs that want to change the Ethereum protocol. It should clearly explain why the existing protocol specification is inadequate to address the problem that the EIP solves. EIP submissions without sufficient motivation may be rejected outright.-->
+Ethereum currently does not have a special place in the state for tracking state counters such as number of transactions or number of accounts.
```

Comment:
Why does it need to be in the state? If a client keeps track of the counters in their database, then shouldn't that be enough?

Reply:
It needs to be in the state so that one does not need to modify things like snapshot synchronisation. If you just imply it is in the database, but not in the state, snapshot sync needs to also include these data. Having it in the state makes implementation of the counters less intrusive